### PR TITLE
Add ChingariIE and ChingariUserIE.

### PR DESCRIPTION
**Pull request by Ashish0804**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractors
- [ ] New feature

---

### Description of your *pull request* and other information

https://chingari.io/

### CODE REVIEW

File: yt_dlp/extractor/chingari.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+                'quality': 10,
+            })
+        self._sort_formats(formats)
+
+        return {
+            'id': id,
+            'title': compat_urllib_parse_unquote_plus(clean_html(post_data.get('caption'))),
+            'description': compat_urllib_parse_unquote_plus(clean_html(post_data.get('caption'))),
+            'duration': media_data.get('duration'),
+            'thumbnail': base_url + media_data.get('thumbnail', ''),
```
Comment (pukkandan) [inline yt_dlp/extractor/chingari.py:44]: if thumbnail key is not present, it points to base url?

File: yt_dlp/extractor/chingari.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+            'like_count': post_data.get('likeCount'),
+            'view_count': post_data.get('viewsCount'),
+            'comment_count': post_data.get('commentCount'),
+            'repost_count': post_data.get('shareCount'),
+            'timestamp': str_to_int(post_data.get('created_at')) // 1000,
```
Comment (pukkandan) [inline yt_dlp/extractor/chingari.py:49]: Do you have to use `str_to_int`? If not,
```suggestion
            'timestamp': int_or_none(post_data.get('created_at'), 1000),
```
If you do, still take care that `None // 1000` doesnt throw error

File: yt_dlp/extractor/chingari.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+            'uploader_id': post_data.get('userId') or author_data.get('_id'),
+            'uploader': author_data.get('name'),
+            'uploader_url': 'https://chingari.io/' + author_data.get('username', ''),
```
Comment (pukkandan) [inline yt_dlp/extractor/chingari.py:52]: similar issue as thumbnail

File: yt_dlp/extractor/chingari.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+        author_data = post_data.get('authorData', {})
+        song_data = post_data.get('song', {})  # revist this in future for differentiating b/w 'art' and 'author'
+
+        formats = []
+        for frmt, frmt_path in media_data.get('transcoded', {}).items():
+            formats.append({
+                'format_id': frmt,
+                'width': str_to_int(frmt[1:]),
+                'url': base_url + frmt_path,
+            })
```
Comment (pukkandan) [inline yt_dlp/extractor/chingari.py:29]:
```suggestion
        formats = [{
            'format_id': frmt,
            'width': str_to_int(frmt[1:]),
            'url': base_url + frmt_path,
        } for frmt, frmt_path in media_data.get('transcoded', {}).items())
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/chingari.py (hunk 1 above)
Comment (Ashish0804) [reply yt_dlp/extractor/chingari.py:44]: Yeah but it shouldn't matter. Thumbnail will always be there if the code is at return statement.

Comment (Ashish0804) [reply yt_dlp/extractor/chingari.py]: Taken care of.

File: yt_dlp/extractor/chingari.py (hunk 2 above)
Comment (Ashish0804) [reply yt_dlp/extractor/chingari.py]: Taken care of.

File: yt_dlp/extractor/chingari.py (hunk 3 above)
Comment (Ashish0804) [reply yt_dlp/extractor/chingari.py]: Taken care of.

File: yt_dlp/extractor/chingari.py (hunk 4 above)
Comment (Ashish0804) [reply yt_dlp/extractor/chingari.py]: Taken care of.

**Pull request merged**